### PR TITLE
MDEV-14448: Ctrl-C should not exit the client

### PR DESCRIPTION
**Pull request by an3l**
This PR is cherry-picked from PR #703 and updated (see commits).

### CODE REVIEW

**Comment by vuvova** [comment]
I don't understand that, why would you need that many changes, while it looks like all you needed was something like 
```
--- a/client/mysql.cc
+++ b/client/mysql.cc
@@ -1398,8 +1398,10 @@ sig_handler handle_sigint(int sig)
   /* terminate if no query being executed, or we already tried interrupting */
   if (!executing_query || (interrupted_query == 2))
   {
-    tee_fprintf(stdout, "Ctrl-C -- exit!\n");
-    goto err;
+    rl_on_new_line(); // Regenerate the prompt on a newline
+    rl_replace_line("", 0); // Clear the previous text
+    rl_redisplay();
+    return;
   }
```
?

Althought it's not a very good solution, it hardcodes the behavior for Ctrl-C (and has a bug for `interrupted_query == 2`). A better solution would be to install sigint handler only for the duration of a query. Like, install the handler instead of `executing_query= 1`, and remove the handler instead of `executing_query= 0`

**Comment by an3l** [comment]
Hi @vuvova ,
I have updated the patch with respect to comments of #703 (mysql patch)
So current behavior is
```
MariaDB [(none)]> Ctrl-C
MariaDB [(none)]> Ctrl-C
MariaDB [(none)]> use testCtrl-C
```
Thanks

File: client/mysql.cc [hunk 1]
```diff
@@ -1408,8 +1408,11 @@ sig_handler handle_sigint(int sig)
...
   if (!executing_query || (interrupted_query == 2))
   {
-    tee_fprintf(stdout, "Ctrl-C -- exit!\n");
-    goto err;
+    tee_fprintf(stdout, "Ctrl-C\n");
+    rl_on_new_line(); // Regenerate the prompt on a newline
+    rl_replace_line("", 0); // Clear the previous text
+    rl_redisplay();
+    return;
   }
```
Comment (vuvova) [inline client/mysql.cc]: the condition and comment say "if no query being executed OR we already tried interrupting".

I think that in the second case it's pretty weird to do what you're doing, there's no prompt to regenerate.

Probably, the correct behavior would be still to quit the client. Or, may be better, to disconnect and show the prompt again.

Comment (an3l) [reply client/mysql.cc]: Updated according to your inputs, thanks!

**Comment by an3l** [comment]
Merged with 396313d301b3567aeadd04 in `5.5`.

**Pull request closed without merging**